src/main/scala/sigmastate/Values.scala

```diff
@@ -522,24 +533,72 @@ object Values {
     override def tpe = SBoolean
   }
 
-  /**
-    * For sigma statements
+  /** Algebraic data type of sigma proposition expressions.
+    * Values of this type are used as values of SigmaProp type of SigmaScript and SigmaDsl
     */
-  trait SigmaBoolean extends NotReadyValue[SBoolean.type] {
-    override def tpe = SBoolean
+  trait SigmaBoolean /*extends NotReadyValue[SBoolean.type]*/ {
+    def tpe = SBoolean
 
-    def fields: Seq[(String, SType)] = SigmaBoolean.fields
     /** This is not used as operation, but rather as data value of SigmaProp type. */
     def opType: SFunc = Value.notSupportedError(this, "opType")
+
+    /** Unique id of the node class used in serialization of SigmaBoolean. */
+    val opCode: OpCode
   }
 
   object SigmaBoolean {
     val PropBytes = "propBytes"
-    val IsProven = "isProven"
-    val fields = Seq(
-      PropBytes -> SByteArray,
-      IsProven -> SBoolean
-    )
+    val IsValid = "isValid"
+    object serializer extends SigmaSerializer[SigmaBoolean, SigmaBoolean] {
```

Comment:
Why not put the serializer into a separate file?

Reply:
It just continues the design used in ErgoBox, AvlTreeData etc. I agreed it should be changed though.